code/modules/power/apc.dm

```diff
@@ -1363,6 +1365,23 @@
 			charging = APC_NOT_CHARGING
 			chargecount = 0
 
+		if(excess >= 2500000 && !shock_proof) // SKYRAT EDIT ADD - APC Arcing
+			var/shock_chance = 5 // 5%
+			if(excess >= 7500000)
+				shock_chance = 15
+			else if(excess >= 5000000)
```

Comment:
Those numbers should be in defines, to help with code readability and ease of modification in the future if needed.